adapter/src/ethereum.rs

```diff
@@ -276,6 +266,49 @@ impl Adapter for EthereumAdapter {
     }
 }
 
+#[derive(Debug, Clone)]
+struct RelayerClient {
+    client: Client,
+    relayer_url: String,
+}
+
+impl RelayerClient {
+    pub fn new(relayer_url: &str) -> Result<Self, reqwest::Error> {
+        let client = Client::builder().build()?;
+
+        Ok(Self {
+            relayer_url: relayer_url.to_string(),
+            client,
+        })
+    }
+
+    pub async fn get_privilege(&self, identity: &[u8; 20]) -> Result<u8, AdapterError> {
+        use reqwest::Response;
+        use std::collections::HashMap;
+
+        let relay_url = format!(
+            "{}/identity/by-owner/{}",
+            self.relayer_url,
+            hex::encode(identity)
```

Comment:
identity is `0x` prefixed

Reply:
Yeah I am still testing it, couldn't do it in Thursday.

Reply:
This is fixed! Also I had to pass the `VerifiedPayload.from` and not the identity here.